### DISCUSSION TITLE
Table 表行内按钮执行操作超过 2 秒时的 BUG

### DISCUSSION THREAD

**Discussion by youthalan** [dotnetcore/BootstrapBlazor#4275] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

操作方法：

![image](https://github.com/user-attachments/assets/09502a42-69d9-42ac-91af-144a14b67fab)

结果：

![image](https://github.com/user-attachments/assets/db5b4eaa-f10e-455e-bfb5-e6f5157116d7)

### Expected Behavior

希望这个勾不要被勾选。因为记录都已经移动了其它位置

**Comment by bb-auto** [comment]
@youthalan Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Reply by youthalan** [reply]
项目工程：
[BootstrapBlazorApp1_20240911164113.zip](https://github.com/user-attachments/files/16961207/BootstrapBlazorApp1_20240911164113.zip)

**Comment by ArgoZhang** [comment]
### 带小勾效果

https://github.com/user-attachments/assets/4b15b60d-03d9-4c48-aec2-0232c9bc318d

### 不带小勾效果

https://github.com/user-attachments/assets/65631c5e-937d-4187-9056-21f6baa19129

**Reply by ArgoZhang** [reply]
两种模式都可以，组件没问题

**Reply by ArgoZhang** [reply]
duplicate #4269

**Reply by ArgoZhang** [reply, marked as answer]
解决工程上传

[BootstrapBlazorApp1.zip](https://github.com/user-attachments/files/16971929/BootstrapBlazorApp1.zip)

### 解决问题核心代码

```
<TableCellPopConfirmButton Color="context.Status == 0? Color.Success : Color.Warning" Icon="fa-solid fa-pen"
   Text="@Localizer[(context.Status == 0? "Entry":"Export")]"
   AutoRenderTableWhenClick="true"
   AutoSelectedRowWhenClick="false"
   OnConfirm="@(() => OnEntrance(context))" />
```

增加 `AutoRenderTableWhenClick` `AutoSelectedRowWhenClick` 参数设置

```
private async Task OnEntrance(Foo item)
{
    item.Status = 1 - item.Status;
    item.UpdateDateTime = DateTime.Now;
    await Task.Delay(1000); 
    //await table.QueryAsync();
}
```

移除方法内查询语句
